arangod/Cluster/UpdateReplicatedLogAction.cpp

```diff
@@ -0,0 +1,100 @@
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Lars Maier
+////////////////////////////////////////////////////////////////////////////////
+
+#include <optional>
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/Exceptions.h"
+#include "Basics/StringUtils.h"
+#include "Cluster/MaintenanceFeature.h"
+#include "Network/NetworkFeature.h"
+#include "Replication2/ReplicatedLog/AgencyLogSpecification.h"
+#include "Replication2/ReplicatedLog/Algorithms.h"
+#include "Replication2/ReplicatedLog/NetworkAttachedFollower.h"
+#include "Replication2/ReplicatedLog/ReplicatedLog.h"
+#include "RestServer/DatabaseFeature.h"
+#include "UpdateReplicatedLogAction.h"
+#include "Utils/DatabaseGuard.h"
+
+using namespace arangodb::basics;
+using namespace arangodb::replication2;
+
+
+
+bool arangodb::maintenance::UpdateReplicatedLogAction::first() {
+
+  struct LogActionContextMaintenance : algorithms::LogActionContext {
+    LogActionContextMaintenance(TRI_vocbase_t& vocbase, network::ConnectionPool* pool)
+        : vocbase(vocbase), pool(pool) {}
+
+    auto dropReplicatedLog(LogId id) -> arangodb::Result override {
+      return vocbase.dropReplicatedLog(id);
+    }
+    auto ensureReplicatedLog(LogId id)
+        -> std::shared_ptr<replicated_log::ReplicatedLog> override {
+      return vocbase.ensureReplicatedLog(id, std::nullopt);
+    }
+    auto buildAbstractFollowerImpl(LogId id, ParticipantId participantId)
+    -> std::shared_ptr<replicated_log::AbstractFollower> override {
+      return std::make_shared<replicated_log::NetworkAttachedFollower>(pool, std::move(participantId), vocbase.name(), id);
+    }
+
+    TRI_vocbase_t& vocbase;
+    network::ConnectionPool* pool;
+  };
+
+  auto spec = std::invoke([&]() -> std::optional<agency::LogPlanSpecification> {
+    auto buffer = StringUtils::decodeBase64(_description.get(REPLICATED_LOG_SPEC));
+    auto slice = VPackSlice(reinterpret_cast<uint8_t const*>(buffer.c_str()));
+    if (!slice.isNone()) {
+      return agency::LogPlanSpecification(agency::from_velocypack, slice);
+    }
+
+    return std::nullopt;
+  });
+
+  auto logId = LogId{StringUtils::uint64(_description.get(REPLICATED_LOG_ID))};
+  auto serverId = ServerState::instance()->getId();
+  auto rebootId = ServerState::instance()->getRebootId();
+
+  network::ConnectionPool* pool = _feature.server().getFeature<NetworkFeature>().pool();
+
+  auto const& database = _description.get(DATABASE);
+  auto& df = _feature.server().getFeature<DatabaseFeature>();
+  DatabaseGuard guard(df, database);
+  auto ctx = LogActionContextMaintenance{guard.database(), pool};
+  auto result =
+      replication2::algorithms::updateReplicatedLog(ctx, serverId, rebootId, logId,
+                                                    spec.has_value() ? &spec.value() : nullptr);
+
+  if (result.fail()) {
+    LOG_TOPIC("ba775", ERR, Logger::REPLICATION2)
```

Comment:
```suggestion
    LOG_TOPIC("ba775", WARN, Logger::REPLICATION2)
```

Reply:
I think we should stick to `ERR` because this is something the user act upon, i.e. no space left or such errors.